### PR TITLE
Remove GL from deviceinfo

### PR DESCRIPTION
**Pull request by hevrard**

### CODE REVIEW

**Comment by hevrard** [comment]
Waiting for the kokoro builds on windows and mac to try on these platforms.

File: core/os/device/deviceinfo/cc/BUILD.bazel [hunk 1]
```diff
@@ -43,13 +40,10 @@ cc_library(
         "//tools/build:linux": [],
         "//tools/build:darwin": [
             "-framework Cocoa",
```
Comment (pmuetschard) [inline core/os/device/deviceinfo/cc/BUILD.bazel]: Do we need Cocoa still?

**Comment by chrisforbes** [comment]
Hold on -- I'm seeing some unexplained weirdness in my trial merge of this.

File: core/os/device/deviceinfo/cc/BUILD.bazel (hunk 1 above)
Comment (hevrard) [reply core/os/device/deviceinfo/cc/BUILD.bazel]: I guess not?

I don't have a mac to try it out. Could you have a quick go at removing this and check that it compiles?

Also, do we still need `-lgdi32` for windows?

**Comment by chrisforbes** [comment]
```
ERROR: /src/gapid/gapii/cc/BUILD.bazel:32:1: apic compiling apis for arm64-v8a failed (Segmentation fault): apic failed: error executing command 
  (cd /home/chrisforbes/.cache/bazel/_bazel_chrisforbes/e62b479852179e01aa2b7bfc0560144e/sandbox/linux-sandbox/10103/execroot/gapid && \
  exec env - \
    PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
  bazel-out/host/bin/cmd/apic/linux_amd64_stripped/apic compile --search .,bazel-out/android-arm64-v8a-fastbuild/bin --target arm64-v8a --capture '' --module '' --output bazel-out/android-arm64-v8a-fastbuild/bin/gapii/cc/apis_compiled.o '--optimize=True' '--dump=False' --namespace gapii --symbols c++ --emit-encode gapis/api/vulkan/vulkan.api)
Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox apic failed: error executing command 
  (cd /home/chrisforbes/.cache/bazel/_bazel_chrisforbes/e62b479852179e01aa2b7bfc0560144e/sandbox/linux-sandbox/10103/execroot/gapid && \
  exec env - \
    PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
  bazel-out/host/bin/cmd/apic/linux_amd64_stripped/apic compile --search .,bazel-out/android-arm64-v8a-fastbuild/bin --target arm64-v8a --capture '' --module '' --output bazel-out/android-arm64-v8a-fastbuild/bin/gapii/cc/apis_compiled.o '--optimize=True' '--dump=False' --namespace gapii --symbols c++ --emit-encode gapis/api/vulkan/vulkan.api)
Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox
Target //:pkg failed to build

```

**Comment by hevrard** [comment]
@chrisforbes I don't see which part of this change would impact apic and lead to such errors... do you have any idea?

**Comment by lpy** [comment]
It can be compiled when I merged this patch locally. I think we are good to go.

File: core/os/device/deviceinfo/cc/windows/query.cpp [hunk 2]
```diff
@@ -72,68 +36,13 @@ void destroyContext() {
...
-  }
-  if (gContext.mGlCtx != nullptr) {
-    gContext.wglMakeCurrent(gContext.mHDC, 0);
-    gContext.wglDeleteContext(gContext.mGlCtx);
-  }
-}
-
-void createGlContext() {
-  gContext.wglCreateContext =
-      (pfn_wglCreateContext)core::GetGlesProcAddress("wglCreateContext");
```
Comment (AWoloszyn) [inline core/os/device/deviceinfo/cc/windows/query.cpp]: Removing all calls to this function will likely break stuff, as this function is where we keep a hack to work around broken MESA drivers.

**Comment by AWoloszyn** [comment]
There are some related hacks in "gapii/vulkan/vk_graphics_spy/cc/layer.cpp" that should be removed now.

Also we can actually entirely remove the second `.so` file that exists now that we only support a single API.

**Comment by AWoloszyn** [comment]
These hacks are also needed for tracing. So when if you remove it from apic, it may not work.

**Comment by hevrard** [comment]
This is on hold until we try protozero.

If protozero can handle our protos and leads to acceptable capture performances, we can remove our dependency on LLVM. Then we won't have to care about Mesa LLVM symbol leaks.

**Comment by hevrard** [comment]
Probably made obsolete by #208

**Comment by pmuetschard** [comment]
Yes, #208 supersedes this.

**Pull request closed without merging**